java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/binder/TimestampWithPrecisionBinder.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.arrow.adapter.jdbc.binder;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.sql.Types;
+import java.util.Calendar;
+import org.apache.arrow.vector.TimeStampVector;
+import org.apache.arrow.vector.TimeStampWithPrecisionVector;
+import org.apache.arrow.vector.types.pojo.ArrowType;
+
+/** A column binder for timestamps. */
+public class TimestampWithPrecisionBinder extends BaseColumnBinder<TimeStampVector> {
+  private final Calendar calendar;
+  private final long unitsPerSecond;
+  private final long nanosPerUnit;
+
+  /** Create a binder for a timestamp vector using the default JDBC type code. */
+  public TimestampWithPrecisionBinder(TimeStampWithPrecisionVector vector, Calendar calendar) {
+    this(
+        vector,
+        calendar,
+        isZoned(vector.getField().getType()) ? Types.TIMESTAMP_WITH_TIMEZONE : Types.TIMESTAMP);
+  }
+
+  /**
+   * Create a binder for a timestamp vector.
+   *
+   * @param vector The vector to pull values from.
+   * @param calendar Optionally, the calendar to pass to JDBC.
+   * @param jdbcType The JDBC type code to use for null values.
+   */
+  public TimestampWithPrecisionBinder(
+      TimeStampWithPrecisionVector vector, Calendar calendar, int jdbcType) {
+    super(vector, jdbcType);
+    this.calendar = calendar;
+
+    final ArrowType.TimestampWithPrecision type =
+        (ArrowType.TimestampWithPrecision) vector.getField().getType();
+    switch (type.getPrecision()) {
+      case 0:
+        this.unitsPerSecond = 1;
```

Comment:
Why this have to be a member of the class? I feel these should be part of some kind of static lookup list.